### DISCUSSION TITLE
Use AirByte to manage loading data from github repos

### DISCUSSION THREAD

**Discussion by bilalshaikh42** [biosimulations/biosimulations#3005] [Ideas]
https://airbyte.io/
We can use github connector or make connector

**Comment by jonrkarr** [comment]
Is airbyte about the content of repos or the metadata in the GitHub API?

How would airbyte be used?

I was thinking we use GitHub actions to push content -- whose primary home is a GitHub repo -- to BioSimulations via the BioSimulations API. I think we have all of the ingredients already. One thing that would make it a little easier is a reusable GitHub action (i.e., actions from the GitHub Action Marketplace) for pushing content to BioSimulations.

**Comment by bilalshaikh42** [comment]
I was thinking about it as a potential way to manage all the workflows/repeated jobs of importing various repositories in one place. Looking at it more closely, it does seem the connector is for the metadata. 
Something similar that could be used is something like [Argo Workflows ](https://argoproj.github.io/argo-workflows/)

**Reply by bilalshaikh42** [reply]
The advantage to this over just using github actions is being able to mange all the import jobs in one place/give others access to the workflows. If we end up not needing those, then actions are just fine

**Comment by jonrkarr** [comment]
Fundamentally, I think we should aim for the control of the workflows to be decentralized so that model repositories can manage this as they see best fit. I think we should think of this similar to how we've viewed simulators -- something that the community contributes.

Of course, at the beginning we're really doing most of the contributions ourselves. If we maintain a centralized design, it would stay that way. If we aim for a more decentralized approach, community contributions would hopefully emerge over time. I think this is important for community buy in and scale. If we try to do everything ourselves, we'll be limited by the amount of funding we can obtain.

That said, I agree it would be helpful to factor out any common code and workflow steps. Other aspects of strategies include
- Factoring out common code to a common Python library (we're already doing this)
- Create a GitHub action step for the GitHub action marketplace
